File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        Map<Character, Integer> map=new HashMap<>();
        int i=0, j=0, maxL=0, maxF=0;
        while(j<s.length()){
            char c=s.charAt(j);
            map.put(c, map.getOrDefault(c,0)+1);
            maxF=Math.max(maxF,map.get(c));
            if((j-i+1)-maxF>k){
                map.put(s.charAt(i),map.get(s.charAt(i))-1);
                i++;
            }
            maxL=Math.max(maxL,j-i+1);
            j++;
        }
        return maxL;
    }
}

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l=0
        temp=k
        maxLen=0
        seen={}
        max_freq=0
        for r in range(len(s)):
            seen[s[r]]=seen.get(s[r],0)+1
            max_freq=max(max_freq, seen.get(s[r]))
            if r-l+1-max_freq>k:
                seen[s[l]]-=1
                l+=1
            maxLen=max(maxLen, r-l+1)
        return maxLen